bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+            zone = channelUID.getGroupId();
+            switch (channelWithoutGroup) {
+                case CHANNEL_POWER:
+                    if (command.equals(OnOffType.ON)) {
+                        httpResponse = setPower("on", zone);
+                        tmpString = getResponseCode(httpResponse);
+                        if (!tmpString.equals("0")) {
+                            updateState(channelUID, OnOffType.OFF);
+                        }
+                    } else if (command.equals(OnOffType.OFF)) {
+                        httpResponse = setPower("standby", zone);
+                        tmpString = getResponseCode(httpResponse);
+                        if (!tmpString.equals("0")) {
+                            updateState(channelUID, OnOffType.ON);
+                        }
+                    }
+                    break;
+                case CHANNEL_MUTE:
+                    if (command.equals(OnOffType.ON)) {
+                        httpResponse = setMute("true", zone);
+                        tmpString = getResponseCode(httpResponse);
+                        if (!tmpString.equals("0")) {
+                            updateState(channelUID, OnOffType.OFF);
+                        }
+                    } else if (command.equals(OnOffType.OFF)) {
+                        httpResponse = setMute("false", zone);
+                        tmpString = getResponseCode(httpResponse);
+                        if (!tmpString.equals("0")) {
+                            updateState(channelUID, OnOffType.ON);
+                        }
+                    }
+                    break;
+                case CHANNEL_VOLUME:
+                    tmpString = command.toString();
+                    tmpString = tmpString.replace(".0", "");
+                    try {
+                        volumePercent = Integer.parseInt(tmpString);
+                        volumeAbsValue = (maxVolumeState * volumePercent) / 100;
+                        setVolume(volumeAbsValue, zone, this.host);
+                        if (config.configSyncVolume) {
+                            tmpString = getDistributionInfo(this.host);
+                            @Nullable
+                            DistributionInfo targetObject = new Gson().fromJson(tmpString, DistributionInfo.class);
+                            role = targetObject.getRole();
+                            if (role.equals("server")) {
+                                for (JsonElement ip : targetObject.getClientList()) {
+                                    JsonObject clientObject = ip.getAsJsonObject();
+                                    setVolumeLinkedDevice(volumePercent, zone,
+                                            clientObject.get("ip_address").getAsString());
+                                }
+                            }
+                        } // END configSyncVolume
+                    } catch (Exception e) {
```

Comment:
You should specify the concrete exception you want to catch, since catching `Exception` catches also all runtime exceptions. Please check all.

Reply:
Removed the exception check as it was no longer necessary